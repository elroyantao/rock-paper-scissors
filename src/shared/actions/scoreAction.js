export const playerWins = () => ({
  type: 'PLAYER_WINS'
})

export const computerWins = () => ({
  type: 'COMPUTER_WINS'
})

export const itsATie = () => ({
  type: 'ITS_A_TIE'
})

export const resetScore = () => ({
  type: 'RESET_SCORE'
})
