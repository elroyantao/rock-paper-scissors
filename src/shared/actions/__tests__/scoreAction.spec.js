import * as actions from '../scoreAction'

const snapshot = (action) => expect(action).toMatchSnapshot()

describe('scoreAction', () => {
  it('playerWins', () => {
    snapshot(actions.playerWins())
  })
  it('computerWins', () => {
    snapshot(actions.computerWins())
  })
  it('itsATie', () => {
    snapshot(actions.itsATie())
  })
  it('resetScore', () => {
    snapshot(actions.resetScore())
  })
})
