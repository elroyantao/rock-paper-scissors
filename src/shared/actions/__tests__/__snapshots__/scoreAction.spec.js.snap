// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scoreAction computerWins 1`] = `
Object {
  "type": "COMPUTER_WINS",
}
`;

exports[`scoreAction itsATie 1`] = `
Object {
  "type": "ITS_A_TIE",
}
`;

exports[`scoreAction playerWins 1`] = `
Object {
  "type": "PLAYER_WINS",
}
`;

exports[`scoreAction resetScore 1`] = `
Object {
  "type": "RESET_SCORE",
}
`;
